adapter/internal/discovery/xds/server.go

```diff
@@ -599,7 +695,22 @@ func IsAPIExist(vhost, name, version string) (exists bool) {
 
 // GenerateIdentifierForAPI generates an identifier unique to the API
 func GenerateIdentifierForAPI(vhost, name, version string) string {
-	return vhost + ":" + name + ":" + version
+	return fmt.Sprint(vhost, apiKeyFieldSeparator, name, apiKeyFieldSeparator, version)
+}
+
+// GenerateIdentifierForAPIWithoutVhost generates an identifier unique to the API name and version
+func GenerateIdentifierForAPIWithoutVhost(name, version string) string {
+	return fmt.Sprint(name, apiKeyFieldSeparator, version)
+}
+
+// ExtractVhostFromAPIIdentifier extracts vhost from the API identifier
+func ExtractVhostFromAPIIdentifier(id string) string {
+	elem := strings.Split(id, apiKeyFieldSeparator)
+	if len(elem) == 3 {
+		return elem[0]
+	}
+	// TODO: (renuka) check this to have handle (return error?)
+	return "*"
```

Comment:
Handle this to drop the API deployment in adapter.

Reply:
handled with https://github.com/wso2/product-microgateway/pull/1826/commits/3aa9b7de891d72fbd25fdfb61f7dac23d0f11d99